### PR TITLE
Small tweaks

### PR DESCRIPTION
**Pull request by TiagoCavalcante**

### CODE REVIEW

File: mathics/builtin/inference.py [hunk 1]
```diff
@@ -273,9 +272,6 @@ def get_assumption_rules_dispatch(evaluation):
     if val_consistent_assumptions == SymbolFalse:
         evaluation.message("Inconsistent assumptions")
 
-    if assumptions_list is None:
```
Comment (mmatera) [inline mathics/builtin/inference.py:276]: Not sure about this one.

File: mathics/builtin/inference.py [hunk 2]
```diff
@@ -298,10 +294,8 @@ def get_assumption_rules_dispatch(evaluation):
 
         if value:
             symbol_value = SymbolTrue
-            symbol_negate_value = SymbolFalse
```
Comment (mmatera) [inline mathics/builtin/inference.py:301]: Sure that these are not necesary? I do not remember exactly why I put these lines...

**Comment by mmatera** [comment]
Apart from the comments, LGTM

File: mathics/builtin/inference.py (hunk 1 above)
Comment (TiagoCavalcante) [reply mathics/builtin/inference.py:276]: `pred` isn't defined in this function, so if it be `None`, an undeclared variable exception would be raised.

File: mathics/builtin/inference.py (hunk 2 above)
Comment (mmatera) [reply mathics/builtin/inference.py:301]: In that case, go ahead...

File: mathics/builtin/inference.py (hunk 1 above)
Comment (mmatera) [reply mathics/builtin/inference.py:276]: OK, if the CI does not complains, then go ahead. At some point, I have to go through this code again anyway

**Comment by rocky** [comment]
Benchmarks, please? Any PR that purports to improving performance should have benchmarks attached. Thanks.

**Comment by mmatera** [comment]
This was just cleaning, I think it is OK. In any case, I will try to merge this on #13 to see if there are no conflicts.

**Pull request merged**